### ISSUE TITLE
Remove bespoke retry logic in `rpc` crate

### ISSUE THREAD

**Issue by frisitano** [0xPolygonZero/zero-bin#102]
Following the migration from `ethers` to `alloy` in the `rpc` create we needed to implement a bespoke retry layer for the alloy provider.  We should remove this once a retry mechanism is shipped upstream in `alloy`.

Reference: https://github.com/fractal-zkp/zero-bin/blob/2cbcf98ea8bf57ee17775d2e72c51dd40e8cd04b/rpc/src/retry.rs#L1-L148

**Comment by Nashtare** [comment]
Is this worth a feature-request upstream?

**Comment by frisitano** [comment]
> Is this worth a feature-request upstream?

Yes, I've added comment in the PR I got insight from - https://github.com/alloy-rs/examples/pull/95
